test/EndToEndTests/Tests/Core/Microsoft.OData.Core.E2E.Tests/WriteJsonPayloadTests/WritePayloadTests.cs

```diff
@@ -0,0 +1,1132 @@
+        {
+            var settings = new ODataMessageWriterSettings
+            {
+                ODataUri = new ODataUri() { ServiceRoot = _baseUri },
+                EnableMessageStreamDisposal = false
+            };
+            string outputWithModel = null;
+            string outputWithoutModel = null;
+
+            var responseMessageWithModel = new TestStreamResponseMessage(new MemoryStream());
+            responseMessageWithModel.SetHeader("Content-Type", mimeType);
+
+            var orderType = _model.FindDeclaredType(NameSpacePrefix + "Order") as IEdmEntityType;
+            var orderSet = _model.EntityContainer.FindEntitySet("Orders");
+
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithModel, settings, _model))
+            {
+                var odataWriter = await messageWriter.CreateODataResourceSetWriterAsync(orderSet, orderType);
+                outputWithModel = await WriteAndVerifyOrderFeedAsync(responseMessageWithModel, odataWriter, true, mimeType);
+            }
+
+            var responseMessageWithoutModel = new TestStreamResponseMessage(new MemoryStream());
+            responseMessageWithoutModel.SetHeader("Content-Type", mimeType);
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithoutModel, settings))
+            {
+                var odataWriter = await messageWriter.CreateODataResourceSetWriterAsync();
+                outputWithoutModel = await WriteAndVerifyOrderFeedAsync(
+                    responseMessageWithoutModel,
+                    odataWriter,
+                    false,
+                    mimeType);
+            }
+
+            //var rex = new Regex("\"\\w*@odata.type\":\"#[\\w\\(\\)\\.]*\",");
+            //outputWithoutModel = rex.Replace(outputWithoutModel, "");
+            //WritePayloadHelper.VerifyPayloadString(outputWithModel, outputWithoutModel, mimeType);
+            if (mimeType == MimeTypes.ApplicationJson + MimeTypes.ODataParameterFullMetadata)
+            {
+                var rex = new Regex("\"\\w*@odata.associationLink\":\"[^\"]*\",");
+                var outputWithModel2 = rex.Replace(outputWithModel, "");
+                var outputWithoutModel2 = rex.Replace(outputWithoutModel, "");
+                TestsHelper.VerifyPayloadString(outputWithModel2, outputWithoutModel2, mimeType);
+            }
+            else if (mimeType == MimeTypes.ApplicationJson + MimeTypes.ODataParameterMinimalMetadata)
+            {
+                var rex = new Regex("\"\\w*@odata.type\":\"#[\\w\\(\\)\\.]*\",");
+                var outputWithoutModel2 = rex.Replace(outputWithoutModel, "");
+                TestsHelper.VerifyPayloadString(outputWithModel, outputWithoutModel2, mimeType);
+            }
+            else
+            {
+                //NoMetadata with/out model should result in same output
+                Assert.Equal(outputWithModel, outputWithoutModel);
+            }

```

Comment:
If we can't create suitable parameters to allow us to have a more "linear" and predictable test, maybe we should split the scenario into separate tests.